api/src/main/java/org/jboss/forge/roaster/model/util/Types.java

```diff
@@ -298,7 +298,12 @@ public static boolean isGeneric(String type)
       StringTokenizer tok = new StringTokenizer(typeArgs, ", ");
       while (tok.hasMoreTokens())
       {
-         if (!validateNameWithGenerics(tok.nextToken()))
+         String typeArg = tok.nextToken();
+         while (incompleteGenerics(typeArg) && tok.hasMoreElements()) {
+            typeArg += tok.nextToken();
```

Comment:
Are you sure about this? Since `StringBuilder#toString()` would have to be called for each iteration, wouldn't this negate the performance gain?